<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Karissa Barthelson" />

<meta name="date" content="2022-10-31" />

<title>FeDex_fth1a</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2022_MPSII_qPCR_ironSupplementation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="FeDex_tfr1a.html">FeDex_trf1a</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">FeDex_fth1a</h1>
<h4 class="author">Karissa Barthelson</h4>
<h4 class="date">2022-10-31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2022_MPSII_qPCR_ironSupplementation/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221025code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20221025)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221025code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221025)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarissab2022MPSsocietyironSupplementaiontreea769cbb95489f45b4ce0c6e34ae4794c703c6f2ftargetblanka769cbba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karissa-b/2022_MPS-society_ironSupplementaion/tree/a769cbb95489f45b4ce0c6e34ae4794c703c6f2f" target="_blank">a769cbb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarissab2022MPSsocietyironSupplementaiontreea769cbb95489f45b4ce0c6e34ae4794c703c6f2ftargetblanka769cbba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karissa-b/2022_MPS-society_ironSupplementaion/tree/a769cbb95489f45b4ce0c6e34ae4794c703c6f2f" target="_blank">a769cbb</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/FeDex_fth1a.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(readxl)
library(here)
library(magrittr)
library(scales)
library(readxl)
library(ggpubr)
library(ggeasy)
library(ggrepel)
library(ggfortify)
library(ggbeeswarm)
library(ggforce)
library(kableExtra)
library(emmeans)
library(performance)
library(see) # performance dependency
library(car)

# set the default theme for ggplot as theme_bw
theme_set(theme_bw())</code></pre>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Here, we are assessing the level of <em>fth1a</em> transcripts in the brains of 6 month old zebradfish. These zebrafish are either heterozygous (unaffected) or homozygous (MPS-IIIB) for the A603fs mutatin in <em>naglu</em>. They were treated with either vehicle (0.85% saline), 0.5ug iron-dextran or 1 ug of iron-dextran. Then, they were allowed to recover alone in a pair mate tank for either 7 or 14 days. Here, we will determine whether there are significant changes to the expression of <em>fth1a</em> with any of the combinations described.</p>
<p>Ewan did this by qPCR (using 96 well plates). The plot below summarises the number of fish in the study.</p>
<pre class="r"><code># read in data
data &lt;- read_xlsx(&quot;data/fth1a plate 1-7 analysis 1.xlsx&quot;) %&gt;% 
    mutate(fish_id = as.character(fish_id), 
         treatment = factor(treatment, 
                            levels = c(&quot;saline&quot;, 
                                       &quot;0.5 ug iron-dextran&quot;, 
                                       &quot;1 ug iron-dextran&quot;))
    )

# define the names of the columns which have metadata in it. 
# will be helpful when selecting cols later
metacols &lt;- colnames(data)[1:8]</code></pre>
<p>The plot below summarizes the metadata of the fish we tested here.</p>
<pre class="r"><code>data %&gt;% 
  dplyr::select(metacols) %&gt;% 
  dplyr::filter(Plate %in% seq(1:4)) %&gt;% # only keep first 4 plates to strart with. talk about the others later
  group_by(genotype, sex, treatment, dpi) %&gt;% 
  mutate(n = n()) %&gt;% 
  ggplot(aes(x = genotype)) +
  geom_bar(aes(fill = treatment), 
           position = &quot;dodge&quot;, 
           colour = &quot;black&quot;) +
  facet_wrap(~dpi+sex, nrow = 1) +
  scale_fill_viridis_d() + 
  theme(legend.position = &quot;bottom&quot;)+
  labs(title = &quot;Number of fish per genotype, sex, treatment and dpi in the qPCR analysis&quot;,
       y = &quot;Number of fish&quot;, 
       x = &quot;&quot;)</code></pre>
<p><img src="figure/FeDex_fth1a.Rmd/unnamed-chunk-3-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
<p>Becuase there was so many fish, ewan had to split up the samples across multiple plates.</p>
<pre class="r"><code>data %&gt;% 
  dplyr::select(metacols) %&gt;% 
  dplyr::filter(Plate %in% seq(1:4)) %&gt;% # only keep first 4 plates to strart with. talk about the others later
  group_by(genotype, sex, treatment, dpi, Plate) %&gt;% 
  mutate(n = n(), 
         Plate = paste(&quot;Plate&quot;, Plate)) %&gt;% 
  ggplot(aes(x = genotype)) +
  geom_bar(aes(fill = treatment), 
           position = &quot;dodge&quot;, 
           colour = &quot;black&quot;) +
  facet_wrap(~dpi+Plate, nrow = 1) +
  scale_fill_viridis_d() + 
  theme(legend.position = &quot;bottom&quot;)+
  labs(title = &quot;Number of fish per genotype, sex, treatment and dpi in the qPCR analysis&quot;,
       y = &quot;Number of fish&quot;, 
       x = &quot;&quot;)</code></pre>
<p><img src="figure/FeDex_fth1a.Rmd/unnamed-chunk-4-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="check-technical-replicates-of-actin-housekeeping-gene" class="section level1">
<h1>Check technical replicates of actin (housekeeping gene)</h1>
<p>The following plot below shows the technical replicates of the actin gene. There seem to be some dodgy ones, particularly those on Plate 4, likely due to pipetting error or some contamination of the RNA. The technical replicate which appears to be the problematic one will be omitted.</p>
<pre class="r"><code># plot technical replicates for actin by plate. 
data %&gt;% 
  dplyr::filter(Keep == &quot;Y&quot;) %&gt;% 
  dplyr::filter(Plate %in% c(1, 2, 3, 4)) %&gt;% # only look at the ones ewan did first. Will get to plates 5-7 later
  dplyr::select(all_of(metacols), starts_with(&quot;actin&quot;)) %&gt;% 
  gather(key = &quot;actin&quot;, value = &quot;ct&quot;, starts_with(&quot;actin ct&quot;)) %&gt;% 
  dplyr::filter(ct != &quot;Undetermined&quot;) %&gt;% 
  mutate(fish_id = factor(fish_id, levels = c(&quot;NTC&quot;, seq(1:100))), # put in order
         ct = as.numeric(ct), 
         actin = str_remove(actin, pattern = &quot;actin ct &quot;) 
         ) %&gt;% 
  group_by(fish_id) %&gt;% 
  mutate(var = sd(ct)) %&gt;% 
  ggplot(aes(x = fish_id, y = ct)) + 
  geom_point() +
  geom_boxplot(fill = NA) +
  geom_label_repel(aes(label = paste0(fish_id, &quot;_&quot;, actin)), # only label the variable fish
                   data = . %&gt;% 
                     dplyr::filter(var &gt; 1)) +
  facet_wrap(~Plate, scales = &quot;free_x&quot;, nrow = 1) +
  ggtitle(&quot;actin technical replicates&quot;, 
          subtitle = &quot;fish are labelled if the standard deviation of the tech reps is &gt; 1&quot;)</code></pre>
<p><img src="figure/FeDex_fth1a.Rmd/unnamed-chunk-5-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="check-technical-replicates-of-fth1a" class="section level1">
<h1>Check technical replicates of fth1a</h1>
<p>Like for actin, the technical replicates for plate 3 and 4 look dodgy. We think something has happened during those PCRs.</p>
<pre class="r"><code># plot technical replicates for fth1a by plate. 
data %&gt;% 
  dplyr::filter(Keep == &quot;Y&quot;) %&gt;% 
  dplyr::filter(Plate %in% c(1, 2, 3, 4)) %&gt;% # only look at the ones ewan did first. Will get to plates 5-7 later
  dplyr::select(all_of(metacols), starts_with(&quot;fth1a&quot;)) %&gt;% 
  gather(key = &quot;fth1a&quot;, value = &quot;ct&quot;, starts_with(&quot;fth1a ct&quot;)) %&gt;% 
  dplyr::filter(ct != &quot;Undetermined&quot;) %&gt;% 
  mutate(fish_id = factor(fish_id, levels = c(&quot;NTC&quot;, seq(1:100))), # put in order
         ct = as.numeric(ct), 
         actin = str_remove(fth1a, pattern = &quot;fth1a ct &quot;) 
         ) %&gt;% 
  group_by(fish_id) %&gt;% 
  mutate(var = sd(ct)) %&gt;% 
  ggplot(aes(x = fish_id, y = ct)) + 
  geom_point() +
  geom_boxplot(fill = NA) +
  geom_label_repel(aes(label = paste0(fish_id, &quot;_&quot;, fth1a)), # only label the variable fish
                   data = . %&gt;% 
                     dplyr::filter(var &gt; 1)) +
  facet_wrap(~Plate, scales = &quot;free_x&quot;, nrow = 1) +
  ggtitle(&quot;fth1a technical replicates&quot;, 
          subtitle = &quot;fish are labelled if the standard deviation of the tech reps is &gt; 1&quot;)</code></pre>
<p><img src="figure/FeDex_fth1a.Rmd/unnamed-chunk-6-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="repeating-plates-3-4" class="section level1">
<h1>Repeating plates 3 + 4</h1>
<p>Ewan repeated the qPCRs which were on plates 3 and 4 (now 5 and 6) A lot of variability is still obsereved for actin (which theoretically should be similar across samples as it is a house-keeping gene).</p>
<pre class="r"><code>ggarrange(
  # actin
data %&gt;% 
  dplyr::filter(Keep == &quot;Y&quot;) %&gt;% 
  dplyr::filter(Plate %in% c(1, 2, 5, 6, 7)) %&gt;% # only look at the ones ewan did first. Will get to plates 5-7 later
  dplyr::select(all_of(metacols), starts_with(&quot;actin&quot;)) %&gt;% 
  gather(key = &quot;actin&quot;, value = &quot;ct&quot;, starts_with(&quot;actin ct&quot;)) %&gt;% 
  dplyr::filter(ct != &quot;Undetermined&quot;) %&gt;% 
  mutate(fish_id = factor(fish_id, levels = c(&quot;NTC&quot;, seq(1:100))), # put in order
         ct = as.numeric(ct), 
         Plate = paste(&quot;Plate&quot;, Plate), 
         actin = str_remove(actin, pattern = &quot;actin ct &quot;) 
         ) %&gt;% 
  group_by(fish_id) %&gt;% 
  mutate(var = sd(ct)) %&gt;% 
  ggplot(aes(x = fish_id, y = ct)) + 
  geom_point() +
  geom_boxplot(fill = NA) +
  geom_label_repel(aes(label = paste0(fish_id, &quot;_&quot;, actin)), # only label the variable fish
                   data = . %&gt;% 
                     dplyr::filter(var &gt; 1)) +
  facet_wrap(~Plate, scales = &quot;free_x&quot;, nrow = 1) +
  ggtitle(&quot;actin technical replicates&quot;), 

# fth1a
data %&gt;% 
  dplyr::filter(Keep == &quot;Y&quot;) %&gt;% 
  dplyr::filter(Plate %in% c(1, 2, 5, 6, 7)) %&gt;% # only look at the ones ewan did first. Will get to plates 5-7 later
  dplyr::select(all_of(metacols), starts_with(&quot;fth1a&quot;)) %&gt;% 
  gather(key = &quot;fth1a&quot;, value = &quot;ct&quot;, starts_with(&quot;fth1a ct&quot;)) %&gt;% 
  dplyr::filter(ct != &quot;Undetermined&quot;) %&gt;% 
  mutate(fish_id = factor(fish_id, levels = c(&quot;NTC&quot;, seq(1:100))), # put in order
         ct = as.numeric(ct), 
         Plate = paste(&quot;Plate&quot;, Plate), 
         fth1a = str_remove(fth1a, pattern = &quot;fth1a ct &quot;) 
         ) %&gt;% 
  group_by(fish_id) %&gt;% 
  mutate(var = sd(ct)) %&gt;% 
  ggplot(aes(x = fish_id, y = ct)) + 
  geom_point() +
  geom_boxplot(fill = NA) +
  geom_label_repel(aes(label = paste0(fish_id, &quot;_&quot;, fth1a)), # only label the variable fish
                   data = . %&gt;% 
                     dplyr::filter(var &gt; 1)) +
  facet_wrap(~Plate, scales = &quot;free_x&quot;, nrow = 1) +
  ggtitle(&quot;fth1a technical replicates&quot;), 
ncol = 1
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/FeDex_fth1a.Rmd/unnamed-chunk-7-1.png" alt="fish are labelled if the standard deviation of the tech reps is &gt; 1" width="100%" height="100%" />
<p class="caption">
fish are labelled if the standard deviation of the tech reps is &gt; 1
</p>
</div>
</div>
<div id="plot-the-results" class="section level1">
<h1>Plot the results</h1>
<p>The expression of fth1a appears to increase with increasing amounts of iron dose in the hets, and apparently appears to increase in the homs. However, there seems to be a substansial batch effect of <code>Plate</code>, particularly Plate 6. It is not confounded (i.e. Plate 6 contains only 14dpi, homs + hets, males + female, all treatments). However, not enough of each metadata group is in this plate to faithfully remove this batch effect.</p>
<pre class="r"><code># Plot the results 
data_clean &lt;- data %&gt;% 
  dplyr::filter(Keep == &quot;Y&quot;, 
                Plate %in% c(1, 2, 5, 6, 7)) %&gt;%  # only keep the ones ewan repeated
  gather(key = &quot;gene&quot;, value = &quot;ct&quot;, contains(&quot; ct &quot;)) %&gt;% 
  mutate(fish_id = as.factor(fish_id), 
         ct = as.numeric(ct),  
         Plate = as.factor(Plate),
         techrep = str_extract(gene, pattern = &quot;[1|2|3]&quot;), 
         gene = str_remove(gene, pattern = &quot; ct [1|2|3]&quot;) 
  ) %&gt;% 
   dplyr::filter(ct != &quot;Undetermined&quot;) %&gt;% 
  # filter the problematic techreps
  dplyr::filter(
    fish_id != 27, 
    !(gene == &quot;fth1a&quot; &amp; fish_id == 27 &amp; techrep == 3),
    !(gene == &quot;fth1a&quot; &amp; fish_id == 7 &amp; techrep == 3),

    !(gene == &quot;actin&quot; &amp; fish_id == 34 &amp; techrep == 1)
    ) %&gt;% 
  group_by(fish_id, gene) %&gt;% 
  mutate(aveCT = mean(ct)) %&gt;% 
  dplyr::select(-techrep, -ct) %&gt;% 
  unique() %&gt;% 
  spread(gene, aveCT) %&gt;% 
  mutate(dct = fth1a - actin ) %&gt;% 
  mutate(treatment = factor(treatment, 
                            levels = c(&quot;saline&quot;, 
                                       &quot;0.5 ug iron-dextran&quot;, 
                                       &quot;1 ug iron-dextran&quot;))
         
  )</code></pre>
<pre class="r"><code>data_clean %&gt;% 
ggplot(aes(x = genotype, y = dct, fill = genotype)) +
  geom_boxplot(aes(fill = genotype), 
               outlier.shape = NA) + 
  geom_point(aes(shape = Plate), size =3) +
  scale_fill_viridis_d(begin = 0.5) +
  facet_wrap(~dpi+treatment, nrow = 1) </code></pre>
<p><img src="figure/FeDex_fth1a.Rmd/unnamed-chunk-9-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # stat_compare_means(aes(label = paste0(&quot;p = &quot;, ..p.format..) )) </code></pre>
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<p>While some subtle changes to expression of <em>fth1a</em> might be present after i.p. injection of iron-dextran, the batch effect of plate is too substantial to be confident. Therefore, Ewan will attempt to repeat this experiment, doing all qPCRs at once using a 384 well plate, removing the batch effect completely.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] car_3.0-11         carData_3.0-4      see_0.7.3          performance_0.10.0
 [5] emmeans_1.6.1      kableExtra_1.3.4   ggforce_0.3.3      ggbeeswarm_0.6.0  
 [9] ggfortify_0.4.12   ggrepel_0.9.1      ggeasy_0.1.3       ggpubr_0.4.0      
[13] scales_1.1.1       magrittr_2.0.1     here_1.0.1         readxl_1.3.1      
[17] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.7        purrr_0.3.4       
[21] readr_1.4.0        tidyr_1.1.3        tibble_3.1.2       ggplot2_3.3.6     
[25] tidyverse_1.3.1   

loaded via a namespace (and not attached):
 [1] TH.data_1.0-10    colorspace_2.0-2  ggsignif_0.6.2    ellipsis_0.3.2   
 [5] rio_0.5.27        rprojroot_2.0.2   estimability_1.3  fs_1.5.0         
 [9] rstudioapi_0.13   farver_2.1.0      fansi_0.5.0       mvtnorm_1.1-2    
[13] lubridate_1.7.10  xml2_1.3.2        codetools_0.2-18  splines_4.0.2    
[17] knitr_1.33        polyclip_1.10-0   jsonlite_1.7.2    workflowr_1.6.2  
[21] broom_0.7.8       dbplyr_2.1.1      compiler_4.0.2    httr_1.4.2       
[25] backports_1.2.1   assertthat_0.2.1  Matrix_1.3-4      cli_3.0.0        
[29] later_1.2.0       tweenr_1.0.2      htmltools_0.5.1.1 tools_4.0.2      
[33] coda_0.19-4       gtable_0.3.0      glue_1.4.2        Rcpp_1.0.7       
[37] cellranger_1.1.0  jquerylib_0.1.4   vctrs_0.3.8       svglite_2.0.0    
[41] insight_0.18.6    xfun_0.24         openxlsx_4.2.4    rvest_1.0.0      
[45] lifecycle_1.0.0   rstatix_0.7.0     MASS_7.3-54       zoo_1.8-9        
[49] hms_1.1.0         promises_1.2.0.1  sandwich_3.0-1    yaml_2.2.1       
[53] curl_4.3.2        gridExtra_2.3     sass_0.4.0        stringi_1.6.2    
[57] highr_0.9         zip_2.2.0         rlang_0.4.11      pkgconfig_2.0.3  
[61] systemfonts_1.0.2 evaluate_0.14     lattice_0.20-44   labeling_0.4.2   
[65] cowplot_1.1.1     tidyselect_1.1.1  R6_2.5.0          generics_0.1.0   
[69] multcomp_1.4-17   DBI_1.1.1         pillar_1.6.1      haven_2.4.1      
[73] foreign_0.8-81    withr_2.4.2       survival_3.2-11   abind_1.4-5      
[77] modelr_0.1.8      crayon_1.4.1      utf8_1.2.1        rmarkdown_2.9    
[81] grid_4.0.2        data.table_1.14.0 git2r_0.28.0      reprex_2.0.0     
[85] digest_0.6.27     webshot_0.5.2     xtable_1.8-4      httpuv_1.6.1     
[89] munsell_0.5.0     beeswarm_0.4.0    viridisLite_0.4.0 vipor_0.4.5      
[93] bslib_0.2.5.1    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
